spec/index.bs

```diff
@@ -5231,12 +5271,13 @@ called the compute pass encoder can no longer be used.
                     error and stop.
                     <div class=validusage>
                         - |this|.{{GPUProgrammablePassEncoder/[[debug_group_stack]]}}.length is 0.
+
+                        Issue: Add remaining validation.
                     </div>
 
-                Issue: Add remaining validation.
+                1. Push each command in |this|.{{GPUProgrammablePassEncoder/[[command_list]]}} onto the end of
+                    |this|.{{GPUProgrammablePassEncoder/[[command_encoder]]}}.{{GPUCommandEncoder/[[command_list]]}}.
```

Comment:
I think it's probably okay to miss that nuance here. The nuance comes in the definitions of the passes, in particular that load and store happen at the beginning/end of render passes (and that there's no synchronization of shared resources within a render pass - only of rendering results).

Compute passes, OTOH, don't do that much. Synchronization within passes is implicit and I don't think anything in particular happens at the start/end of a compute pass. But we keep them to reflect native APIs better and because intermixing with copy commands could have additional cost (state restoration in particular I think). And because we might extend them someday in such a way that it matters (e.g. opting out of automatic synchronization within the pass).

I think things end up looking a lot simpler than they are because of the closure-capturing thing. In practice state setting is integral to the underlying APIs, and important for most or all hardware (especially non-bindless hardware). We're basically describing things as if they're bindless underneath.

Reply:
Unrelated, just to make sure I didn't miss anything: WebGPU doesn't currently allow for bindless techniques on the part of the app, right?

Reply:
Correct.